### PR TITLE
Use new style for more tables

### PR DESCRIPTION
**Pull request by farodin91**
<details><summary>Screenshots</summary><p>
Log:

Screenshot 2021-11-17 at 12-55-12 Log  Jenkins

System Info:
Screenshot 2021-11-17 at 12-55-05 System Information  Jenkins

About:
image

Jenkins CLI:
image
</p>
</details>

### Test this pull request

```
docker run --rm -ti -p 8080:8080 -e ID=5925 -e MERGE_WITH=master jenkins/core-pr-tester
```

### Proposed changelog entries

* Modernise System Info and Log Recorder pages.
* Display full user name, rather than id, in `/securityRealm/` when using the built-in security realm (regression since 2.xxx).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by farodin91** [comment]
@timja Would you like to review?

**Comment by timja** [comment]
@janfaracik can you take a look? (I've sent an invite to you so I can request you review next time)

**Review by fqueiruga — APPROVED** [review]
CSS and markup LGTM, don;t know about that JS

**Comment by farodin91** [comment]
@fqueiruga I just format the file to use 2 spaces instead of random.

**Review by uhafner — APPROVED** [review]
The conversion looks good, thanks!

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by timja — APPROVED** [review]

**Comment by uhafner** [comment]
Out of scope for this PR: while looking at the new layout I wonder how we can improve the layout of such pages in general, do you have an idea @janfaracik? 

The table now looks like a Bootstrap card, but the actual title of the page (e.g. "Log Recorders") somehow stands out of this card on the top without any "connection" to the table. It would be awesome if we can combine both parts into a single concept. Does someone have an idea how to change that?

**Comment by timja** [comment]
> Out of scope for this PR: while looking at the new layout I wonder how we can improve the layout of such pages in general, do you have an idea @janfaracik?
> 
> The table now looks like a Bootstrap card, but the actual title of the page (e.g. "Log Recorders") somehow stands out of this card on the top without any "connection" to the table. It would be awesome if we can combine both parts into a single concept. Does someone have an idea how to change that?

I have more of an issue with all the clipboards and the button touching the table.

**Comment by janfaracik** [comment]
Hey, looking good :) 

So a lot of the new table stuff/design stuff isn't documented yet and that's totally my bad, I'll hopefully have something started and up around December.

A few things from me:

- For the headers (inside of `jenkins-app-bar`) I'd remove the large clipboard icons, they don't add anything to the page really and the size/colour/prominence of them detracts from the actual content of page - feel free to disagree though :)
- For the links in the table, I'd recommend using the class `jenkins-table__link` - this will give you the rounded box hover effects that the other tables have
- For the icons in the table, I'd recommend replacing them with icons from https://ionic.io/ionicons - `clipboard-outline` would look great instead of the clipboard as would `settings-outline` for the settings button. To use these icons, download them from the website and drag them into `/war/src/main/resources/images`, then reference them like so `<l:ionicon name="settings-outline" />` (see the Manage Nodes page for an example)
- To style the buttons in the table (eg the settings button) use:
```
<td class="jenkins-table__cell--tight">
  <div class="jenkins-table__cell__button-wrapper">
    <a href="${rootURL}/${c.url}configure" class="jenkins-table__button">
      <l:ionicon name="settings-outline" />
    </a>
   </div>
</td>
```

Thanks for the changes :)

File: core/src/main/resources/jenkins/model/Jenkins/systemInfo.jelly
Comment (janfaracik) [inline core/src/main/resources/jenkins/model/Jenkins/systemInfo.jelly:36]: Theres a little bit of a spacing issue with the headings on this page but not a blocker imo, will be easy to resolve once the spacing classes are merged in from https://github.com/jenkinsci/jenkins/pull/5916#issuecomment-968352585

**Comment by uhafner** [comment]
> I have more of an issue with all the clipboards and the button touching the table. and the settings button being so far away isn't great either

Yes, in the logger view this should be changed indeed. Maybe it would make sense to move the configuration link to the first column? And remove the clipboard icon totally as it makes no sense.

**Comment by uhafner** [comment]
> * For the headers (inside of `jenkins-app-bar`) I'd remove the large clipboard icons, they don't add anything to the page really and the size/colour/prominence of them detracts from the actual content of page - feel free to disagree though :)

Yes, we should either use an icon on all page headers or in none of them. I also would remove them here.

> * For the links in the table, I'd recommend using the class `jenkins-table__link` - this will give you the rounded box hover effects that the other tables have
> * For the icons in the table, I'd recommend replacing them with icons from https://ionic.io/ionicons - `clipboard-outline` would look great instead of the clipboard as would `settings-outline` for the settings button. To use these icons, download them from the website and drag them into `/war/src/main/resources/images`, then reference them like so `<l:ionicon name="settings-outline" />` (see the Manage Nodes page for an example)

👍 

(I'm not sure whether the icon should be a gear (configure) or a pencil (edit) symbol?)

> * For the button in the table (eg the settings button), I'd recommend styling the cell and button like so:
> 
> ```
> <td class="jenkins-table__cell--tight">
>   <div class="jenkins-table__cell__button-wrapper">
>     <a href="${rootURL}/${c.url}configure" class="jenkins-table__button">
>       <l:ionicon name="settings-outline" />
>     </a>
>    </div>
> </td>
> ```
> 
> Thanks for the changes :)

**Comment by janfaracik** [comment]
> Out of scope for this PR: while looking at the new layout I wonder how we can improve the layout of such pages in general, do you have an idea @janfaracik?
> 
> The table now looks like a Bootstrap card, but the actual title of the page (e.g. "Log Recorders") somehow stands out of this card on the top without any "connection" to the table. It would be awesome if we can combine both parts into a single concept. Does someone have an idea how to change that?

I agree, things definitely feel a little disconnected right now. I'm not too sure what best way to approach it yet, I think one thing that's really missing from Jenkins' design right now is some meaningful hierarchy, there's quite a lot of content/controls on each page but little order/structure to them.

I'm hopeful that once more of the changes are merged in things will start to feel more cohesive.

**Comment by fqueiruga** [comment]
@janfaracik I think you're right. One thing I wanted to do as well was bringing the ui samples plugin back into core and have some sort of storybook-like experience where devs could look at proper uses of the jelly widgets.

**Comment by farodin91** [comment]
image

I could also update about page in this PR, if wanted?

**Comment by timja** [comment]
go for it

**Comment by farodin91** [comment]
Jenkins cli
image

**Comment by timja** [comment]
There's the manage users page if you want to keep going? 😄

**Comment by farodin91** [comment]
@timja done
image

File: core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly
Comment (timja) [inline core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly:55]: there's an ionicon for this check the people page, accessible from home screen

File: core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly
Comment (janfaracik) [inline core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly:75]: If you could add `jenkins-table__button--red` to the classes for this button it'll make the trash can red once #5916 is in :) Thanks!

File: core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly
Comment (farodin91) [reply core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly:55]: Yeah, i thought we should use this, otherwise you want see the custom icon.

File: core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly
Comment (farodin91) [reply core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly:75]: Should i switch to close-circle-outline?

Comment (janfaracik) [reply core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly:75]: That's a good question, I think for now it's fine to leave as the trash can as my current line of thinking is that the act of uninstalling/removing a plugin is different to deleting an entity, although that may change once we start updating more areas of Jenkins 👍

Comment (uhafner) [reply core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly]: I'm also trying these new styles in my plugin but I am somewhat unfamiliar with the new syntax we are using here. I tried to use such a button in small by adding the suffix `--small` but that seems to be not possible: is this not correct: `jenkins-table__button--small`. Or do those new button hover effects work only in tables?

Comment (timja) [reply core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly]: doesn't look like they are restricted to tables https://github.com/janfaracik/jenkins/blob/update-plugin-manager/war/src/main/less/modules/table.less

File: core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly
Comment (timja) [inline core/src/main/resources/hudson/security/HudsonPrivateSecurityRealm/index.jelly:63]: This was broken in https://github.com/jenkinsci/jenkins/pull/4803 but may as well fix it.

image

```suggestion
                <a href="${user.url}/" class="jenkins-table__link">${user.fullName}</a>
```

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
> I'm hopeful that once more of the changes are merged in things will start to feel more cohesive.

Isn't it going to be the opposite as soon as you have plugins installed?

IIUC, everyone who shows a table on a view needs to adapt it to get the new style? Plugin maintainers (at least historically) hesitate to increase the core dependency too quickly to allow users on older releases to receive updates. That means they cannot use new core APIs (and I guess this amounts to that on the UI side) in their plugins. Not to mention that many well-working plugins are not very actively maintained, and changes like this need someone to implement them.

**Pull request merged**
